server/src/main/java/org/elasticsearch/upgrades/SystemIndexMigrator.java

```diff
@@ -465,6 +485,7 @@ private void migrateSingleIndex(ClusterState clusterState, Consumer<BulkByScroll
                                 e
                             );
                             removeReadOnlyBlockOnReindexFailure(oldIndex, delegate2, e);
+                            delegate2.onFailure(e); // TODO ensure this is correct.
```

Comment:
@gwbrown Is it correct that I added this here?  I think we'd want to handle this with `onFailure`, otherwise the cleanup logic won't be invoked.  Just making sure I'm not missing something, thanks!